### ISSUE TITLE
Fatal error: Declaration of Magium\Util\Log\Logger::addError()

### ISSUE THREAD

**Issue by Jignesh001** [magium/MagiumMagentoExample#8]
Hi,

I am very new to magiumlib and I have followed the instruction here to begin with first magento test: https://magiumlib.com/blog/how-to-start-selenium-testing-on-magento-or-succeeding-at-selenium-in-under-10-minutes/

It works perfect if there are no error or warning on test but its through fatal error(see below for exact msg) when there are failure in test.
Fatal error: Declaration of Magium\Util\Log\Logger::addError() must be compatible with PHPUnit_Framework_TestListener::addError(PHPUnit_Framework_Test $test, Exception $e, $time) in /MagiumMagentoExample/vendor/magium/magium/lib/Util/Log/Logger.php on line 17

Can you please help me to fix it?

Kind Regards,
Jignesh

**Comment by kschroeder** [comment]
@jig150784 Could you provide the full stack trace?

**Comment by Jignesh001** [comment]
I am using terminal to run phpunit so I don't have much stack trace but here is what i get on debug.

$ php phpunit-5.7.26.phar --debug -c MagiumMagentoExample/tests/phpunit.xml.dist 
PHPUnit 5.7.26 by Sebastian Bergmann and contributors.


Starting test 'MyTests\Admin\LoginTest::testLogin'.
.
Starting test 'MyTests\Admin\NavigateToOrderTest::testOrderPageAccessible'.

Fatal error: Declaration of Magium\Util\Log\Logger::addError() must be compatible with PHPUnit_Framework_TestListener::addError(PHPUnit_Framework_Test $test, Exception $e, $time) in /Users/patelj/Jigneshp/work/selenium/MagiumMagentoExample/vendor/magium/magium/lib/Util/Log/Logger.php on line 17

**Comment by kschroeder** [comment]
Is it possible to use PHPUnit 6?  I will work to resolve this, but this is most likely related to a compatibility layer I have for running either PHPUnit 5 or 6.

**Comment by Jignesh001** [comment]
I am using PHP 5.6 so I have to use PHPUnit 5.7.
As you said, I have tried with the PHPUnit 6 but it says version unsupported.
"php phpunit-6.5.5.phar --stop-on-failure -c MagiumMagentoExample/tests/phpunit.xml.dist 
PHPUnit 6.5.5 by Sebastian Bergmann and contributors.
This version of PHPUnit is supported on PHP 7.0 and PHP 7.1.
You are using PHP 5.6.30 (/usr/bin/php)."


**It works perfect when there is no error.** 
php phpunit-5.7.26.phar --stop-on-failure -c MagiumMagentoExample/tests/phpunit.xml.dist 
PHPUnit 5.7.26 by Sebastian Bergmann and contributors.
.                                                                   1 / 1 (100%)
Time: 5.64 seconds, Memory: 17.00MB
OK (1 test, 5 assertions)

Problem is only when there are some issues on test:
Fatal error: Declaration of Magium\Util\Log\Logger::addError() must be compatible with PHPUnit_Framework_TestListener::addError(PHPUnit_Framework_Test $test, Exception $e, $time) in /Users/patelj/Jigneshp/work/selenium/MagiumMagentoExample/vendor/magium/magium/lib/Util/Log/Logger.php on line 17

**Comment by kschroeder** [comment]
Makes sense.  Try locking your PHPUnit version for the project's `composer.json` file.  For example, `composer require phpunit/phpunit:^5.7`.  That *should* shift the compatibility layer to PHPUnit 5.
